papis/commands/serve.py

```diff
@@ -455,33 +413,33 @@ def clear_cache(self, libname: str) -> None:
     def page_tags(self, libname: Optional[str] = None) -> None:
         libname = libname or papis.api.get_lib_name()
         docs = papis.api.get_all_documents_in_lib(libname)
-        tags_of_tags = [get_tag_list(d["tags"]) for d in docs]
+        tags_of_tags = [ensure_tags_list(d["tags"]) for d in docs]
         tags = sorted(set(tag
                           for _tags in tags_of_tags
                           for tag in _tags))
-        tag_renderer = functools.partial(render_tag, libname=libname)
-        page = TAGS_PAGE.format(tags="".join(map(tag_renderer, tags)),
-                                libname=libname,
-                                placeholder="tags")
-        self.wfile.write(bytes(page, "utf-8"))
+        page = _tags(libname=libname,
+                     pretitle="TAGS",
+                     tags=tags)
+        self.wfile.write(bytes(str(page), "utf-8"))
         self.wfile.flush()
 
     @ok_html
     def page_libraries(self) -> None:
-        page = render_libraries()
-        self.wfile.write(bytes(page, "utf-8"))
+        libname = papis.api.get_lib_name()
+        page = _libraries(libname=libname)
+        self.wfile.write(bytes(str(page), "utf-8"))
         self.wfile.flush()
 
     @ok_html
     def page_document(self, libname: str, ref: str) -> None:
         docs = papis.api.get_documents_in_lib(libname, ref)
         if len(docs) > 1:
             raise Exception("More than one document match %s", ref)
-        elif len(docs) == 0:
+        if len(docs) == 0:
             raise Exception("No document found with ref %s", ref)
```

Comment:
```suggestion
        if len(docs) == 0:
            raise Exception("No document found with ref '{}'".format(ref))
```
? And in the raise above. Before it should just print the tuple?